darwin/importer/importer.py

```diff
@@ -952,36 +973,55 @@ def import_annotations(  # noqa: C901
         ]
         if files_to_track:
             _warn_unsupported_annotations(files_to_track)
-            for parsed_file in track(files_to_track):
-                image_id, default_slot_name = remote_files[parsed_file.full_path]
-                # We need to check if name is not-None as Darwin JSON 1.0
-                # defaults to name=None
-                if parsed_file.slots and parsed_file.slots[0].name:
-                    default_slot_name = parsed_file.slots[0].name
-
-                metadata_path = is_properties_enabled(parsed_file.path)
-
-                errors, _ = _import_annotations(
-                    dataset.client,
-                    image_id,
-                    remote_classes,
-                    attributes,
-                    parsed_file.annotations,  # type: ignore
-                    default_slot_name,
-                    dataset,
-                    append,
-                    delete_for_empty,
-                    import_annotators,
-                    import_reviewers,
-                    metadata_path,
-                )
 
-                if errors:
-                    console.print(
-                        f"Errors importing {parsed_file.filename}", style="error"
-                    )
-                    for error in errors:
-                        console.print(f"\t{error}", style="error")
+            if use_multi_cpu:
+                with concurrent.futures.ThreadPoolExecutor(
+                    max_workers=cpu_limit
+                ) as executor:
+                    futures = [
+                        executor.submit(import_annotation, file)
+                        for file in files_to_track
+                    ]
+                    for _ in tqdm(
+                        concurrent.futures.as_completed(futures),
+                        total=len(futures),
+                        desc="Importing annotations from local file",
+                    ):
+                        future = next(concurrent.futures.as_completed(futures))
+                        try:
+                            future.result()
+                        except Exception as exc:
+                            console.print(
+                                f"Generated an exception: {exc}", style="error"
+                            )
+            else:
+                for file in tqdm(
+                    files_to_track, desc="Importing annotations from local file"
+                ):
+                    import_annotation(file)
+
+    if use_multi_cpu:
+        with concurrent.futures.ThreadPoolExecutor(max_workers=cpu_limit) as executor:
+            futures = [
+                executor.submit(process_local_file, local_path)
+                for local_path in {local_file.path for local_file in local_files}
+            ]
```

Comment:
There are 2 multi-threading loops (1 inside the other). Both are necessary. This is the outer loop, and it parallelises the import of each local annotation file for import.

This is useful for annotation formats where there is a 1-to-1 mapping between annotation file and dataset item, such as Darwin JSON 2.0